### DISCUSSION TITLE
Does pugixml need to hide pragma checks?

### DISCUSSION THREAD

**Discussion by pachadotdev** [zeux/pugixml#648] [Q&A]
I am curious about this line https://github.com/zeux/pugixml/blob/f1f532acfcd94426d4811bed4a2880657739004e/src/pugixml.cpp#L57

When we use pugixml with R, the R checks say

```
*** checking pragmas in C/C++ headers and code ... NOTE**
File which contains pragma(s) suppressing diagnostics:
  'src/vendor/pugixml.cpp'
```

**Comment by zeux** [comment]
Not sure what you mean by "hide", but push/pop is used to avoid warning configuration leaking into the parent translation unit in case header mode is used (when .cpp file is included instead of being compiled separately).

**Reply by pachadotdev** [reply]
"hide" as in "suppressing"

thanks for your answer

pugixml is amazing, I used it as an alternative to Apache Xerces because I cannot use AX with R

furthermore, pugixml works on Windows, which is more amazing
